lib/github/kv.rb

```diff
@@ -252,6 +253,50 @@ def setnx(key, value, expires: nil)
       }
     end
 
+    # increment :: String, Integer, expires: Time? -> Integer
+    #
+    # Increment the key's value by an amount.
+    #
+    # key     - The key to increment.
+    # amount  - The amount to increment the key's value by.
+    # expires - When the key should expire.
+    #
+    # Returns the key's value after incrementing.
+    def increment(key, amount: 1, expires: nil)
+      sql = GitHub::SQL.run(<<-SQL, key: key, amount: amount, now: now, expires: expires || GitHub::SQL::NULL, connection: connection)
+        INSERT INTO key_values (`key`, `value`, `created_at`, `updated_at`, `expires_at`)
+        VALUES(:key, :amount, :now, :now, :expires)
+        ON DUPLICATE KEY UPDATE
+          `value`=IF(
+            concat('',`value`*1) = `value`,
```

Comment:
This is just a hack to see if the string value is numeric since MySQL doesn't have a builtin for that.

Reply:
Without this check, it looks like we'd just clobber any existing value, right? It might be okay to just accept that behavior.

Reply:
That's what would happen. This check prevents the update from affecting the values. I think it would make sense to throw an error if the value isn't incremental? Might save some headaches down the road.

Reply:
🤷‍♂I don't feel strongly either way. It's added complexity to check this, but you're right that it could catch an unfortunate bug in the future.

Reply:
Let's leave it for now unless further review says otherwise.